js2-mode.el

```diff
@@ -7923,14 +7957,18 @@ leaving a statement, an expression, or a function definition."
 Scanner should be initialized."
   (let ((pos js2-ts-cursor)
         (end js2-ts-cursor)  ; in case file is empty
-        root n tt)
+        root n tt
+        (in-directive-prologue t)
+        (saved-strict-mode js2-in-use-strict-directive)
```

Comment:
I'd rather bind `js2-in-use-strict-directive` here so itself, then it'll save us the `saved-strict-mode` variable, and the value will be restored upon exiting the `let` form.